lib/msal-node/src/network/HttpClient.ts

```diff
@@ -274,3 +276,36 @@ const networkRequestViaHttps = <T>(
     });
 };
 
+/**
+ * Check if extra parsing is needed on the repsonse from the server
+ * @param statusCode {number} the status code of the response from the server
+ * @param headers {Record<string, string>} the headers of the response from the server
+ * @param body {string} the body from the response of the server
+ * @returns {Object} JSON parsed body or error object
+ */
+const parseBody = (statusCode: number, headers: Record<string, string>, body: string) => {
+    /**
+     * If the server encounters an error (500 <= statusCode <=599),
+     * the parsed body will be a string error message instead of the expected ServerAuthorizationTokenResponse.
+     * In this case, extra formatting will need to be performed on the error (500 <= statusCode <=599).
+     */
+    let parsedBody;
+    try {
+        parsedBody = JSON.parse(body);
+    } catch (e) {
+        let error;
+        if ((statusCode >= 500) && (statusCode <= 599)) {
```

Comment:
I think semantically 429 also falls in this category - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429